commons/zenoh-buffers/src/bbuf.rs

```diff
@@ -127,15 +127,15 @@ impl Writer for &mut BBuf {
         self.capacity() - self.len()
     }
 
-    fn with_slot<F>(&mut self, len: usize, f: F) -> Result<NonZeroUsize, DidntWrite>
+    unsafe fn with_slot<F>(&mut self, len: usize, write: F) -> Result<NonZeroUsize, DidntWrite>
```

Comment:
The behavior doesn't exactly match the trait documentation. Wwe provide a buffer of _at least_ and not _exactly_ `len` bytes because `as_writable_slice` returns the entire available slice.

Reply:
True